solvers/fista_automatic_restart.py

```diff
@@ -28,63 +24,66 @@ def set_objective(self, X, y, lmbd, fit_intercept):
         self.X, self.y, self.lmbd = X, y, lmbd
         self.fit_intercept = fit_intercept
 
-    def run(self, n_iter):
+    def run(self, callback):
         L = self.compute_lipschitz_constant()
         n_features = self.X.shape[1]
         w = np.zeros(n_features)
         z = np.zeros(n_features)
-        C = 6.38
-        ite_per_restart = np.zeros(int(n_iter / (2 * C)) + 2)
-        F_tab = np.zeros(int(n_iter / (2 * C)) + 2)
-        i_glob = 0
-        i_int = 0
-        counter = 0  # Restart counter
-        LastF = 0
-        ite_per_restart[counter] = int(2*C)
-        F_tab[counter] = (.5*np.linalg.norm(self.X @ w - self.y, ord=2)**2
-                          + self.lmbd*abs(w).sum())
+        C = 6.38  # Parameter controlling the restart frequency, 6.38 is the
+        # optimal theoretical value (we refer the user to the preprint for
+        # more details).
+        inner_iteration = 0
+        n_restarts = 0  # Restart counter
+        last_F = 0
+        ite_per_restart = np.array([int(2*C)])
+        F_tab = np.array([self.cost_function(w)])
+        cb = callback(w)
```

Comment:
I would rename this variable to be more explicit:
```suggestion
        should_run = callback(w)
```